### PR TITLE
[ENGA-7153][MINOR] Adds CreateBatch with Output

### PR DESCRIPTION
**Pull request by andrewcartwright1**
Adds a method that will create a batch of entities and then return the create entites as a result,
Currently the override of Create returns the number of rows effected.

### CODE REVIEW

**Review by DynamiteAndy — APPROVED** [review]
this is a test

File: src/Dapperer/Repository.cs [hunk 1]
```diff
@@ -124,6 +124,35 @@ public virtual int Create(IEnumerable<TEntity> entities, bool identityInsert)
             return result;
         }
 
+        public virtual IEnumerable<TEntity> CreateBatch(IEnumerable<TEntity> entities)
```
Comment (tsvetanmanev) [inline src/Dapperer/Repository.cs]: In terms of the naming. We already have Create for batches.
Which is the `Create(IEnumerable<TEntity> entities)` - Arguably it should have been named CreateBatch initially.

But now for this new method that returns output. I think we should call it `CreateBatchWithOutput()`.
What do you think?

File: src/Dapperer/QueryBuilders/MsSql/SqlQueryBuilder.cs [hunk 2]
```diff
@@ -213,6 +213,26 @@ private string GetSelectColumns(ICollection<string> additionalTableColumns, ITab
             return sql;
         }
 
+        public string InsertQueryOutputBatch<TEntity>(
```
Comment (tsvetanmanev) [inline src/Dapperer/QueryBuilders/MsSql/SqlQueryBuilder.cs]: `InsertQueryBatchWithOutput`

File: src/Dapperer/QueryBuilders/MsSql/SqlQueryBuilder.cs [hunk 3]
```diff
@@ -396,24 +416,50 @@ private string ToSqlInsertStatement<TEntity>(IEnumerable<TEntity> data, string t
             return ConvertToSqlInsertStatement(data, tableName, columnNames);
         }
 
+        private string ToSqlInsertOutputStatement<TEntity>(IEnumerable<TEntity> data, string tableName, string[] columnNames)
```
Comment (tsvetanmanev) [inline src/Dapperer/QueryBuilders/MsSql/SqlQueryBuilder.cs]: ToSqlInsertWithOutputStatement

File: src/Dapperer/Repository.cs (hunk 1 above)
Comment (tsvetanmanev) [reply src/Dapperer/Repository.cs]: Same renaming applies to methods in the .Async classes

**Review by tsvetanmanev — COMMENTED** [review]
Have you tested the new code with a private version of the Dapperer packages in the other solutions you are currently working on?

**Review by tsvetanmanev — APPROVED** [review]

**Pull request merged**